src/ansys/simai/core/api/mixin.py

```diff
@@ -199,25 +180,27 @@ def download_file(
             output_file = file
             close_file = False
 
-        request_kwargs = {"stream": True, "return_json": False}
+        request_kwargs = {}
         if request_json_body is not None:
             request_kwargs.update({"json": request_json_body})
-        response = self._request(request_method, download_url, **request_kwargs)
-        if monitor_callback is not None:
-            monitor_callback(int(response.headers.get("Content-Length", 0)))
-        logger.info("Starting download.")
+        logger.debug(f"Request stream {request_method} on {download_url}")
+        full_url = self.build_full_url_for_endpoint(download_url)
         try:
-            for chunk in response.iter_content(chunk_size=1024):
-                bytes_read_delta = output_file.write(chunk)
+            with httpx.stream(request_method, full_url, **request_kwargs) as response:
```

Comment:
```suggestion
            with self._session.stream(request_method, full_url, **request_kwargs) as response:
```
Otherwise it will return HTTP code 401

Reply:
Woaw that's a nice catch ! Thanks